httpx/httpx.go

```diff
@@ -0,0 +1,31 @@
+package httpx
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+
+	"github.com/zenledger-io/zazen/telemetry"
+)
+
+const (
+	jsonContentType = "application/json; charset=utf-8"
+)
+
+// Writeable is an interface for anything that can
+// be written to an http.ResponseWriter.
+type Writeable interface {
+	GetStatusCode() int
+	GetPayload() any
+}
+
+// WriteJSON serializes a Writeable to an http.ResponseWriter.
+func WriteJSON(ctx context.Context, w http.ResponseWriter, ww Writeable) {
```

Comment:
You can make this method as part of the interface and have `Response` struct implement it. Might be useful if we add other types of responses like `HeadResponse`

Reply:
If we did this then we'd need to either
* duplicate the functionality between `Error` and `Response` (which just gives us what we had before)
* leave this function as-is and create two new methods on `Error` and `Response` that delegate to this function

I don't see much value in either approach.